compiler/noirc_frontend/src/hir/resolution/errors.rs

```diff
@@ -678,6 +680,13 @@ impl<'a> From<&'a ResolverError> for Diagnostic {
                     *span)
 
             }
+            ResolverError::ExpectedTrait { span  } => {
+                Diagnostic::simple_error(
+                    "Expected a trait".to_string(), 
```

Comment:
I think this is a bit too general - I could imagine being confused by this as a user. Can you add the "but found ___" part back to this error?

Reply:
I think the "but found {found}" was showing the token, but I'll craft a message here based on the type.

Reply:
Done!